gravitee-apim-rest-api/gravitee-apim-rest-api-service/src/main/java/io/gravitee/apim/infra/query_service/plan/PlanQueryServiceImpl.java

```diff
@@ -0,0 +1,79 @@
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.gravitee.apim.infra.query_service.plan;
+
+import io.gravitee.apim.core.exception.NotFoundDomainException;
+import io.gravitee.apim.core.exception.TechnicalDomainException;
+import io.gravitee.apim.core.plan.query_service.PlanQueryService;
+import io.gravitee.apim.infra.adapter.PlanAdapter;
+import io.gravitee.definition.model.DefinitionVersion;
+import io.gravitee.repository.exceptions.TechnicalException;
+import io.gravitee.repository.management.api.ApiRepository;
+import io.gravitee.repository.management.api.PlanRepository;
+import io.gravitee.repository.management.model.Api;
+import io.gravitee.repository.management.model.Plan;
+import io.gravitee.rest.api.model.v4.plan.GenericPlanEntity;
+import io.gravitee.rest.api.service.exceptions.ApiNotFoundException;
+import io.gravitee.rest.api.service.exceptions.TechnicalManagementException;
+import java.util.List;
+import java.util.Objects;
+import java.util.Optional;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.context.annotation.Lazy;
+import org.springframework.stereotype.Service;
+
+@Service
+public class PlanQueryServiceImpl implements PlanQueryService {
+
+    private final PlanRepository planRepository;
+    private final ApiRepository apiRepository;
+    private static final Logger logger = LoggerFactory.getLogger(PlanQueryServiceImpl.class);
+
+    public PlanQueryServiceImpl(@Lazy final PlanRepository planRepository, ApiRepository apiRepository) {
+        this.planRepository = planRepository;
+        this.apiRepository = apiRepository;
+    }
+
+    @Override
+    public List<GenericPlanEntity> findAllByGeneralConditionsAndIsActive(String apiId, String pageId) {
+        try {
+            return planRepository
+                .findByApi(apiId)
+                .stream()
+                .filter(page ->
+                    Objects.equals(page.getGeneralConditions(), pageId) &&
+                    !(Plan.Status.CLOSED == page.getStatus() || Plan.Status.STAGING == page.getStatus())
+                )
+                .map(this::mapToGeneric)
+                .toList();
+        } catch (TechnicalException e) {
+            logger.error("An error occurred while finding plans by API ID {}", apiId, e);
+            throw new TechnicalDomainException("An error occurred while trying to find plans by API ID: " + apiId, e);
+        }
+    }
+
+    // TODO: Instead of mapping to GenericPlanEntity, map to a model in Core
+    private GenericPlanEntity mapToGeneric(final Plan plan) {
+        try {
+            Optional<Api> apiOptional = apiRepository.findById(plan.getApi());
```

Comment:
we should not do queries in a map method.
There are few impacts in the context of your PR because the stream is filtered and will contain really few elements.
Nonetheless, this method is likely to be reusable.
In my opinion you should pass the full API as a parameter.

Reply:
I am now passing the DefinitionVersion to query in order to correctly map the plan.